explorer/graphql/data/message_provider.go

```diff
@@ -23,3 +26,26 @@ func (p Provider) XMsgCount(ctx context.Context) (*hexutil.Big, bool, error) {
 
 	return &hex, true, nil
 }
+
+func (p Provider) XMsgRange(ctx context.Context, amount uint64, offset uint64) ([]*resolvers.XMsg, bool, error) {
```

Comment:
We should implement this similar to the [Ethereums schema](https://github.com/ethereum/go-ethereum/blob/master/graphql/schema.go#L371) - we need to  have `from` and `to` block height. With the `amount` and `offset` you kind of reveal implementation details. We should change it.
Same applies to `Blocks(from, to)`.